### ISSUE TITLE
Page load errors missing when using v2 script + v2 API

### ISSUE THREAD

**Issue by jiaweihli** [MindscapeHQ/raygun4js#148]
Strangely enough, using the v2 script + v1 API works fine.

I tested manually by trying to reference a non-existent variable (`test.asdf`) in the body of a `$.ready` call in our `app.js`.
I'm configuring Raygun synchronously in `app.js` file (not on ready), which is loaded after our Raygun script block.

Let me know if I can provide any more info.

**Comment by jiaweihli** [comment]
Works (v1):

```javascript
  window.Raygun.init('API_KEY')
    .attach()
```

Doesn't work (v2):
```javascript
  window.rg4js('apiKey', 'API_KEY')
  window.rg4js('enableCrashReporting', true)
```

**Comment by jiaweihli** [comment]
Looks like Pulse doesn't work in v1?

**Comment by CmdrKeen** [comment]
No, Pulse was the reason for needing the V2 API (as it does more work, and always sends data, being more aggressively async was important). 

Are you including the block around the script in V2? It adds a collection and pushes the load errors into it so they can be popped off once the script is ready to send.

**Comment by jiaweihli** [comment]
Yep, this is the Raygun block:

```html
<!-- Raygun tag -->
<script type="text/javascript">
  !function(e,n,t,r,o,s,c,a){e.RaygunObject=o,e[o]=e[o]||function(){(e[o].o=e[o].o||
  []).push(arguments)},s=n.createElement(t),c=n.getElementsByTagName(t)[0],s.async=1,
  s.src=r,c.parentNode.insertBefore(s,c),a=e.onerror,e.onerror=function(n,t,r,s,c)
  {a&&a(n,t,r,s,c),e[o].q=e[o].q||[],e[o].q.push({e:c})}}(window,document,"script",
  "//cdn.raygun.io/raygun4js/raygun.min.js","rg4js");
</script>
<script type="text/javascript">
<% if (locals.email) { %>
  rg4js('setUser', {
    identifier: "test@localhost.com",
    isAnonymous: false,
    email: "test@localhost.com"
  });
<% } %>
</script>
```

**Comment by dwnz** [comment]
Hey, looks like there may be a script order issue here. 

You need to include the snippet at the bottom of your header and the Raygun init lines just before the closing body tag.

Here's a working script as an example

Hope this helps!

```html
<html>
	<head>
		<script src="//code.jquery.com/jquery-1.11.3.min.js"></script>

		<script type="text/javascript">
		!function(a,b,c,d,e,f,g,h){a.RaygunObject=e,a[e]=a[e]||function(){
		(a[e].o=a[e].o||[]).push(arguments)},f=b.createElement(c),g=b.getElementsByTagName(c)[0],
		f.async=1,f.src=d,g.parentNode.insertBefore(f,g),h=a.onerror,a.onerror=function(b,c,d,f,g){
		h&&h(b,c,d,f,g),g||(g=new Error(b)),a[e].q=a[e].q||[],a[e].q.push({
		e:g})}}(window,document,"script","//cdn.raygun.io/raygun4js/raygun.min.js","rg4js");
		</script>
	</head>
	<body>
		<script>
			$(function(){
				bob.test();
			});
		</script>
		<script type="text/javascript">
			rg4js('apiKey', 'YourAPIKey');
			rg4js('enableCrashReporting', true);
			rg4js('enablePulse', true);
			rg4js('setUser',{
				identifier: "test@localhost.com",
				isAnonymous: false,
				email: "test@localhost.com"
			});
		</script>
	</body>
</html>
```

**Comment by fundead** [comment]
This provider has just had a new release which contains fixes and improvements to script config ordering and instantiation, which should improve the issues you were seeing with missing page load errors. If the problem continues after updating to the latest v2.1.0 please open a new issue.

Regards,

Callum Gavin
Raygun Limited

**Comment by jiaweihli** [comment]
Great, upgraded from v1 without changing too much!
